src/GridFS/Bucket.php

```diff
@@ -246,7 +248,9 @@ public function downloadToStream($id, $destination)
             throw InvalidArgumentException::invalidType('$destination', $destination, 'resource');
         }
 
-        stream_copy_to_stream($this->openDownloadStream($id), $destination);
+        if (stream_copy_to_stream($this->openDownloadStream($id), $destination) === false) {
+            throw StreamException::downloadFailed();
```

Comment:
In this case (and the one in `downloadToStreamByName` below, we could print the `uri` associated with the stream retrieved using `stream_get_meta_data()`. Printing the ID can be troublesome, as we don't know the identifier type. While using `(string) $id` would work for scalars and any string able object, this would fail if someone used an array for an ID, as the string cast would then only return "Array" without any further information. This is why I originally skipped adding this information.

However, we can definitely print a filename in `downloadToStreamByName`. I'll make that change and see if I can come up with reliably printing the identifier.

Reply:
I noted that we use `toJson(fromPHP(...))` in `FileNotFoundException`, so I've duplicated that logic 👍

Reply:
Ah, the `uri` is a good idea.

> Printing the ID can be troublesome, as we don't know the identifier type.

IIRC, this came up somewhere else in the driver/library and I've just relied on JSON encoding.

> I noted that we use `toJson(fromPHP(...))` in `FileNotFoundException`

That might have been what I was thinking of.

The URI would also be a good idea, but I'm fine with either and it seems like the ID is more consistent.